server/controllers/ArticleController.js

```diff
@@ -0,0 +1,80 @@
+import Sequelize from 'sequelize';
+import models from '../models';
+
+const { Article, User } = models;
+const { Op } = Sequelize;
+/**
+ * @class ArticleController
+ * @description Article related Operations
+ */
+class ArticleController {
+/**
+ * @description
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} - returns an Article
+ */
+  static getAnArticle(req, res) {
+    const { slug } = req.params;
+    Article.findOne({
+      where: {
+        [Op.or]: [{ slug }, { id: +slug || 0 }]
+      },
+      include: [User]
+    })
+      .then((returnedArticle) => {
+        if (!returnedArticle) {
+          return res.status(404).json({
+            status: 'failure',
+            errors: {
+              message: ['Article not found']
+            }
+          });
+        }
+        const {
+          id,
+          title,
+          description,
+          body,
+          createdAt,
+          updatedAt,
+          userId
+        } = returnedArticle;
+        const {
+          fullName,
+          email,
+          avatarUrl,
+          bio,
+          roleId
+        } = returnedArticle.User;
+        return res.status(200).json({
+          status: 'success',
+          article: {
+            id,
+            title,
+            slug: returnedArticle.slug,
+            description,
+            body,
+            createdAt,
+            updatedAt,
+            userId,
+            author: {
+              fullName,
+              email,
+              avatarUrl,
+              bio,
+              roleId
+            }
+          }
+        });
+      })
+      .catch(err => res.status(500).json({
+        status: 'failure',
+        errors: {
+          message: ['error reading Article\'s table', `${err}`]
```

Comment:
Is this the error we'd be displaying to the user on the frontend?

Reply:
Using  ``` .catch(err => res.status(500).json({
        status: 'failure',
        errors: err.message
      }));``` resolves to these 
     ```{
    "status": "failure",
    "errors": "column \"nan\" does not exist"
}```
while using these ```      .catch(err => res.status(500).json({
        status: 'failure',
        errors: {
          message: ['error reading Article\'s table', `${err}`]
        }
      }));``` resolves to these 
```{
    "status": "failure",
    "errors": {
        "message": [
            "error reading Article's table",
            "SequelizeDatabaseError: column \"nan\" does not exist"
        ]
    }
}```

None of the above errors makes sense to any user on the frontend. If we notice a status 500 error, we could choose to display `Internet connection error` or `server is down`. The above errors are just for development purposes and only makes sense to a developer.